basetest.pm

```diff
@@ -17,6 +17,13 @@ use Mojo::File 'path';
 my $serial_file_pos = 0;
 my $autoinst_log_pos = 0;
 
+my $fail_severity = {
+    '' => 0,
+    'ok' => 100,
+    'softfail' => 200,
+    'fail' => 300
```

Comment:
The mapping needs a few more values, otherwise the PR looks good.

    select distinct(result) from job_modules;
       result   
    ------------
     none
     failed
     pass
     running
     passed
     skipped
     softfailed
     canceled